src/Discord.Net.Core/Entities/Users/IUser.cs

```diff
@@ -112,5 +112,33 @@ public interface IUser : ISnowflakeEntity, IMentionable, IPresence
         ///     contains the DM channel associated with this user.
         /// </returns>
         Task<IDMChannel> GetOrCreateDMChannelAsync(RequestOptions options = null);
+        /// <summary>
+        ///     The flags that are applied to a user's account.
+        /// </summary>
+        /// <remarks>
+        ///     This value is determined by bitwise OR-ing <see cref="UserFlag"/> values together.
+        /// </remarks>
+        /// <returns>
+        ///     The value of flags for this user.
+        /// </returns>
+        UserFlag Flags { get; }
+        /// <summary>
+        ///     The type of Nitro subscription that is active on this user's account.
+        /// </summary>
+        /// <remarks>
+        ///     This information may only be available with the identify OAuth scope,
+        ///     meaning that users and bots will not have access to this information.
+        /// </remarks>
+        /// <returns>
+        ///     The type of Nitro subscription the user subscribes to, or <c>null</c> if this value could not be determined.
+        /// </returns>
+        PremiumType? PremiumType { get; }
+        /// <summary>
+        ///     The user's chosen language option.
+        /// </summary>
+        /// <returns>
+        ///     The value of the user's chosen language option, if provided.
+        /// </returns>
```

Comment:
An example of the returned value would be preferred, since a `string` type is rather generic.